jax/_src/lax/control_flow.py

```diff
@@ -2425,15 +2431,15 @@ def _linear_solve_batching_rule(args, dims, axis_name, main_type, const_lengths,
   return outs, out_dims
 
 
-linear_solve_p = core.Primitive('custom_linear_solve')
+linear_solve_p = core.AxisPrimitive('custom_linear_solve')
 linear_solve_p.multiple_results = True
 linear_solve_p.def_impl(_custom_linear_solve_impl)
 linear_solve_p.def_abstract_eval(_linear_solve_abstract_eval)
 ad.primitive_jvps[linear_solve_p] = _custom_linear_solve_jvp
 xla.initial_style_translations[linear_solve_p] = \
     xla.lower_fun_initial_style(_custom_linear_solve_impl)
 ad.primitive_transposes[linear_solve_p] = _linear_solve_transpose_rule
-batching.initial_style_batchers[linear_solve_p] = _linear_solve_batching_rule
+batching.collective_rules[linear_solve_p] = _linear_solve_batching_rule
```

Comment:
Can we rename `collective_rules` to something else? Perhaps `axis_primitive_batchers`? It's not just about collectives anymore

Reply:
Done.